File: CustomSecuritySample2016/IO/PBIRS/Swagger/Model/SubscriptionHistory.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.PBIRS.Swagger.Model {

  /// <summary>
  /// An object that defines a subscription history or execution record.
  /// </summary>
  [DataContract]
  public class SubscriptionHistory {
    /// <summary>
    /// A unique identifier by which this Subscription History record can be referenced in requests or in other defined objects.
    /// </summary>
    /// <value>A unique identifier by which this Subscription History record can be referenced in requests or in other defined objects.</value>
    [DataMember(Name="Id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Id")]
    public int? Id { get; set; }

    /// <summary>
    /// A unique UUID value that identifies the Subscription that this record is associated with.
    /// </summary>
    /// <value>A unique UUID value that identifies the Subscription that this record is associated with.</value>
    [DataMember(Name="SubscriptionID", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "SubscriptionID")]
    public Guid? SubscriptionID { get; set; }

    /// <summary>
    /// A string value that specifies the type of subscription execution.
    /// </summary>
    /// <value>A string value that specifies the type of subscription execution.</value>
    [DataMember(Name="Type", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Type")]
    public string Type { get; set; }

    /// <summary>
    /// A string that contains the date-time for when the subscription execution started.
    /// </summary>
    /// <value>A string that contains the date-time for when the subscription execution started.</value>
    [DataMember(Name="StartTime", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "StartTime")]
    public DateTime? StartTime { get; set; }

    /// <summary>
    /// A string that contains the date-time for when the subscription execution ended.
    /// </summary>
    /// <value>A string that contains the date-time for when the subscription execution ended.</value>
    [DataMember(Name="EndTime", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "EndTime")]
    public DateTime? EndTime { get; set; }

    /// <summary>
    /// A string value that represents the status of the subscription execution.
    /// </summary>
    /// <value>A string value that represents the status of the subscription execution.</value>
    [DataMember(Name="SubscriptionStatus", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "SubscriptionStatus")]
    public string SubscriptionStatus { get; set; }

    /// <summary>
    /// A string value that contains message that describes the status of the subscription execution.
    /// </summary>
    /// <value>A string value that contains message that describes the status of the subscription execution.</value>
    [DataMember(Name="Message", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Message")]
    public string Message { get; set; }

    /// <summary>
    /// A JSON string value that contains the session information and error details when a subscription execution fails.
    /// </summary>
    /// <value>A JSON string value that contains the session information and error details when a subscription execution fails.</value>
    [DataMember(Name="Details", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "Details")]
    public string Details { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class SubscriptionHistory {\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  SubscriptionID: ").Append(SubscriptionID).Append("\n");
      sb.Append("  Type: ").Append(Type).Append("\n");
      sb.Append("  StartTime: ").Append(StartTime).Append("\n");
      sb.Append("  EndTime: ").Append(EndTime).Append("\n");
      sb.Append("  SubscriptionStatus: ").Append(SubscriptionStatus).Append("\n");
      sb.Append("  Message: ").Append(Message).Append("\n");
      sb.Append("  Details: ").Append(Details).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: CustomSecuritySample2016/scripts/pbirs/src/model/SubscriptionHistory.js
/*
 * Power BI Report Server REST API
 * The Power BI Report Server REST API provides programmatic access to the report server catalog.  For example, basic CRUD operations can be done on folders, reports, KPIs, data sources, datasets, refresh plans, subscriptions, etc.     The REST API can also be used to provide more advanced functionality, such as: * Navigate the folder hierarchy * Discover the contents of a folder * Download a report definition * Modify default report parameters * Change or execute a refresh plan * A whole lot more  The REST API is a RESTful successor to the legacy [SOAP API](https://msdn.microsoft.com/library/reportservice2010.reportingservice2010.aspx).  Since Power BI Report Server is a superset of SQL Server Reporting Services, the Power BI Report Server REST API is a superset of the  [SQL Server Reporting Services REST API](https://app.swaggerhub.com/apis/microsoft-rs/SSRS/2.0).  __Power BI Report Server API Additions__ * January 2019 Release   * /PowerBIReports({Id})/DataModelRoles (GET)   * /PowerBIReports({Id})/DataModelRoleAssignments (GET & PUT)  Happy coding!  __API samples:__ https://github.com/Microsoft/Reporting-Services  __Developer documentation:__ https://powerbi.microsoft.com/documentation/reportserver-developer-handbook-overview/  __Team Blog:__  https://powerbi.microsoft.com/blog/  __Support forums:__  https://community.powerbi.com/t5/Report-Server/bd-p/ReportServer 
 *
 * OpenAPI spec version: 2.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.10
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.PowerBiReportServerRestApi) {
      root.PowerBiReportServerRestApi = {};
    }
    root.PowerBiReportServerRestApi.SubscriptionHistory = factory(root.PowerBiReportServerRestApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The SubscriptionHistory model module.
   * @module model/SubscriptionHistory
   * @version 2.0
   */

  /**
   * Constructs a new <code>SubscriptionHistory</code>.
   * An object that defines a subscription history or execution record.
   * @alias module:model/SubscriptionHistory
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>SubscriptionHistory</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/SubscriptionHistory} obj Optional instance to populate.
   * @return {module:model/SubscriptionHistory} The populated <code>SubscriptionHistory</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('Id'))
        obj.id = ApiClient.convertToType(data['Id'], 'Number');
      if (data.hasOwnProperty('SubscriptionID'))
        obj.subscriptionID = ApiClient.convertToType(data['SubscriptionID'], 'String');
      if (data.hasOwnProperty('Type'))
        obj.type = ApiClient.convertToType(data['Type'], 'String');
      if (data.hasOwnProperty('StartTime'))
        obj.startTime = ApiClient.convertToType(data['StartTime'], 'Date');
      if (data.hasOwnProperty('EndTime'))
        obj.endTime = ApiClient.convertToType(data['EndTime'], 'Date');
      if (data.hasOwnProperty('SubscriptionStatus'))
        obj.subscriptionStatus = ApiClient.convertToType(data['SubscriptionStatus'], 'String');
      if (data.hasOwnProperty('Message'))
        obj.message = ApiClient.convertToType(data['Message'], 'String');
      if (data.hasOwnProperty('Details'))
        obj.details = ApiClient.convertToType(data['Details'], 'String');
    }
    return obj;
  }

  /**
   * A unique identifier by which this Subscription History record can be referenced in requests or in other defined objects.
   * @member {Number} id
   */
  exports.prototype.id = undefined;

  /**
   * A unique UUID value that identifies the Subscription that this record is associated with.
   * @member {String} subscriptionID
   */
  exports.prototype.subscriptionID = undefined;

  /**
   * A string value that specifies the type of subscription execution.
   * @member {module:model/SubscriptionHistory.TypeEnum} type
   */
  exports.prototype.type = undefined;

  /**
   * A string that contains the date-time for when the subscription execution started.
   * @member {Date} startTime
   */
  exports.prototype.startTime = undefined;

  /**
   * A string that contains the date-time for when the subscription execution ended.
   * @member {Date} endTime
   */
  exports.prototype.endTime = undefined;

  /**
   * A string value that represents the status of the subscription execution.
   * @member {module:model/SubscriptionHistory.SubscriptionStatusEnum} subscriptionStatus
   */
  exports.prototype.subscriptionStatus = undefined;

  /**
   * A string value that contains message that describes the status of the subscription execution.
   * @member {String} message
   */
  exports.prototype.message = undefined;

  /**
   * A JSON string value that contains the session information and error details when a subscription execution fails.
   * @member {String} details
   */
  exports.prototype.details = undefined;


  /**
   * Allowed values for the <code>type</code> property.
   * @enum {String}
   * @readonly
   */
  exports.TypeEnum = {
    /**
     * value: "Scheduled"
     * @const
     */
    scheduled: "Scheduled",

    /**
     * value: "OnDemand"
     * @const
     */
    onDemand: "OnDemand"
  };


  /**
   * Allowed values for the <code>subscriptionStatus</code> property.
   * @enum {String}
   * @readonly
   */
  exports.SubscriptionStatusEnum = {
    /**
     * value: "Completed"
     * @const
     */
    completed: "Completed",

    /**
     * value: "InProgress"
     * @const
     */
    inProgress: "InProgress",

    /**
     * value: "Failed"
     * @const
     */
    failed: "Failed"
  };

  return exports;

}));
